sdk/go/pulumi/resource.go

```diff
@@ -314,7 +316,69 @@ func DeleteBeforeReplace(o bool) ResourceOption {
 // DependsOn is an optional array of explicit dependencies on other resources.
 func DependsOn(o []Resource) ResourceOption {
 	return resourceOption(func(ro *resourceOptions) {
-		ro.DependsOn = append(ro.DependsOn, o...)
+		ro.DependsOn = append(ro.DependsOn, func(ctx context.Context) ([]URN, error) {
+			var urns []URN
+			for _, res := range o {
+				directUrn, isKnown, _, err := res.URN().awaitURN(ctx)
```

Comment:
Any dependencies present in the `Output` layer of `res.URN() : URNOutput` are ignored.

Reply:
any particular reason for this? does this match the behavior of the other SDKs?

Reply:
I can test for it. Node can be our reference. I'd need a way to inject deps into the URN output but that perhaps is possible.

Reply:
Matches the behavior of Node. The dependencies specifically on URNOutput do not participate. C# SDK diverges significantly https://github.com/pulumi/pulumi/pull/7611 and I'm taking Node as authority on what should happen.